### ISSUE TITLE
Lane height

### ISSUE THREAD

**Issue by csawp** [esmini/esmini#471]
Can ESMINI support lane height？
[Ex_Pedestrian_Crossing.zip](https://github.com/esmini/esmini/files/12555160/Ex_Pedestrian_Crossing.zip)
![image](https://github.com/esmini/esmini/assets/114643211/9becc549-abf5-4d1f-951c-f886d04995d8)

**Comment by eknabevcc** [comment]
Unfortunately esmini is not yet supporting this nice feature. 

I foresee some work to not only visualize the height difference but also incorporate it in the general z/elevation calculations. I guess it could be added as a simple offset to the elevation profile and any super-elevation. 

It's not high prio, but we'll keep it in mind for the backlog.

BTW: I looked at the attached OpenDRIVE file. Looking at the image I'd expect lane 3 as follows:

```xml
<lane id="3" type="curb" level="false">
    <link>
    </link>
    <width sOffset="0.0000000000000000e+00" a="2.0000000000000001e-01" b="0.0000000000000000e+00" c="0.0000000000000000e+00" d="0.0000000000000000e+00"/>
    <height sOffset="0.0000000000000000e+00" inner="0.000000000000000e-00" outer="1.2000000000000000e-01"/>
</lane>
```
I.e. starting from zero and "climbing" to 0.12m. but maybe it's the tessellation that's "wrong", it should be a vertical edge from lane 2 to 3... What do you think?

**Comment by csawp** [comment]
Thank you for your reply!

Firstly, in my understanding of height, it should have a width to control its slope rate. If it is vertical, the width can be set to 0.0001. The inner of the next lane and the outer of the previous lane are consistent. If there is any inconsistency, the outer should prevail. At least https://odrviewer.io/ It's like this

And then it seems to calculate before the superelevation roll.
![image](https://github.com/esmini/esmini/assets/114643211/52990def-eb78-4bd9-8740-e083d8cad3b2)

**Comment by eknabevcc** [comment]
I agree that the width affects the slope, but only when inner and outer height differ. I mean, what is the point in having both inner and outer height attributes if they have no effect? 

So my understanding is: Of inner and outer height are equal the "lane" become flat. Only if they differ there will be a slope. 

But maybe I miss something or misinterpret inner and outer attributes?

BTW: What is you interpretation of missing height? I assume zero is default (and not height of any neighbor lane).

**Comment by eknabevcc** [comment]
Regarding calculation: I found this in User guide:

"Lane height may be used to elevate a lane independent from the road elevation. Lane height is used to implement small-scale elevation such as raising pedestrian walkways ... Lane height is specified as offset from the road (including elevation, superelevation, shape) in z direction."

Hence it seems that lane height is applied after any road elevation and superelevation, in case it matters...

**Comment by csawp** [comment]
![image](https://github.com/esmini/esmini/assets/114643211/aecc64a4-ae26-4e11-9fdc-db426a3d20ee)
Yes, the part that is too high should be an independent lane, and the red arrow in the figure is a lane with the [LANE_TYPE_CURB] attribute. It should be possible to use rolling to handle this lane.
That's about it. I also modified the XORD file of the case and played the effect on the website
https://odrviewer.io/

**Comment by eknabevcc** [comment]
Good, I think we're aligned regarding the interpretation.  👍

As said, we'll put this in backlog. Would be nice to support this feature. But it requires some work and I don't dare to estimate a finish date. 

Let's keep issue open.
